tests/integration/test_check_downloaded_output.py

```diff
@@ -0,0 +1,68 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from os import path
+import tarfile
+from pathlib import Path
+
+import utils
+
+
+def test_check_downloaded_output(test_env, create_new_request, tmpdir):
+    """
+    Check that the bundle has all the necessities.
+
+    Process:
+    * Send new request to Cachito API
+    * Send request to download appropriate bundle from Cachito
+
+    Checks:
+    * Check that response code is 200
+    * Check that state is "complete"
+    * Check the downloaded data are in gzip format and valid
+    * Check that dir deps/gomod/… contains cached dependencies
+    * Check that dir app/ contains application source code
+    * Check that the same full path filename is not duplicated
+    """
+    response = create_new_request
+    assert response.status == 201
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response = client.wait_for_complete_request(response)
```

Comment:
Optional: It might be nice to include this in the fixture so that all tests that use the fixture don't have to duplicate this.

In essence, the fixture would guarantee that it would supply a Cachito request that is either complete or failed.